src/ui/public/kuery/lib/filter_migration/lib/__tests__/exists.js

```diff
@@ -0,0 +1,35 @@
+import expect from 'expect.js';
+import { convertExistsFilter } from '../exists';
+
+describe('filter to kuery migration', function () {
+
+  describe('exists filter', function () {
+
+    it('should return a kuery node equivalent to the given filter', function () {
+      const filter = {
+        meta: {
+          type: 'exists',
+          key: 'foo',
+        }
+      };
+      const result = convertExistsFilter(filter);
+
+      expect(result).to.have.property('type', 'function');
+      expect(result).to.have.property('function', 'exists');
+      expect(result.arguments[0].value).to.be('foo');
+    });
+
+    it('should return undefined if the given filter is not of type "exists"', function () {
```

Comment:
thanks :smile: just want to point out that even though there is no way in javascript to write down function signatures or even enforce them, the functions still have one.

Personally, as a developer who has to reconstruct those signatures in his head on the fly, I always find `undefined`s a major pain. Every piece of code that works with the results has to check for it and whenever I return to the code I have to derive the meaning of `undefined` again. I appreciate this freeing up of cognitive capacity. :unicorn: